experiments/AMIP/components/atmosphere/climaatmos_extra_diags.jl

```diff
@@ -80,3 +83,95 @@ CAD.add_diagnostic_variable!(
 
     end,
 )
+
+CAD.add_diagnostic_variable!(
+    short_name = "mass_strf",
+    long_name = "Meridional Mass Streamfunction = 2πa cos(lat) ∫_0^z ρv dz",
+    standard_name = "meridional_mass_streamfunction",
+    units = "kg m^-1 s^-1",
+    compute! = (out, state, cache, time) -> begin
+        FT = eltype(state)
+        lat = CC.Fields.coordinate_field(axes(state.f)).lat
+        ρv = CC.Geometry.UVVector.(state.c.uₕ).components.data.:2 .* state.c.ρ
+        ᶠ∫_0_z_ρv = cache.scratch.ᶠtemp_scalar
+        CC.Operators.column_integral_indefinite!(ᶠ∫_0_z_ρv, ρv)
+        twoπa = FT(2π * 6371e3)
+        if isnothing(out)
+            return ᶠ∫_0_z_ρv .* twoπa .* cosd.(lat)
```

Comment:
Should we use `copy(ᶠ∫_0_z_ρv)` here since it's using `cache.scratch`?

Reply:
It should be ok, since after the multiplication the result becomes a new object. Would you agree, @dennisYatunin ?

Reply:
Yeah, it looks like the value in `ᶠ∫_0_z_ρv` is used to set the diagnostic field immediately after it is computed with `column_integral_indefinite!`. Since there's no way its value can change before the diagnostic field is set, you don't need to allocate a copy of it.

Reply:
Nice, thank you for clarifying!